java-frontend/src/main/java/org/sonar/java/se/checks/OptionalGetBeforeIsPresentCheck.java

```diff
@@ -71,10 +75,33 @@ public ProgramState checkPostStatement(CheckerContext context, Tree syntaxNode)
   }
 
   private static List<ProgramState> setNotPresentConstraint(CheckerContext context, Tree syntaxNode) {
```

Comment:
Name of the method is misleading now, because you are setting `PRESENT` constraint in some cases.